### PR TITLE
Fix stimulus presentation errors

### LINKED ISSUE #84: Fix stimulus presentations error in Visual Behavior
These three exceptions result in a `KeyError: nan`, and come from mapping stimulus presentations to colors for plotting. They're probably all related.
- [x] [`physiology/ophys/visual-behavior/VBO-ExperimentData.md`](https://allenswdb.github.io/physiology/ophys/visual-behavior/VBO-ExperimentData.html#plot-example-cell-activity-with-stimuli-and-behavior) Execution Error
- [x] [`physiology/ophys/visual-behavior/VBO-Tutorial-Compare_trial_types.md`](https://allenswdb.github.io/physiology/ophys/visual-behavior/VBO-Tutorial-Compare_trial_types.html#here-is-a-hit-trial) Execution Error
- [x] [`physiology/ophys/visual-behavior/VB-BehaviorSessionData.md`](https://allenswdb.github.io/physiology/ophys/visual-behavior/VB-BehaviorSessionData.html#plot-behavior-data-for-a-portion-of-one-session) Execution Error: different from the one we just fixed. This one is further into the file.

### Build log warnings
```
/opt/databook/databook/physiology/ophys/visual-behavior/VBO-ExperimentData.md: WARNING: Executing notebook failed: CellExecutionError [mystnb.exec]
/opt/databook/databook/physiology/ophys/visual-behavior/VBO-ExperimentData.md: WARNING: Notebook exception traceback saved in: /root/capsule/results/_build/html/reports/physiology/ophys/visual-behavior/VBO-ExperimentData.err.log [mystnb.exec]

/opt/databook/databook/physiology/ophys/visual-behavior/VBO-Tutorial-Compare_trial_types.md: WARNING: Executing notebook failed: CellExecutionError [mystnb.exec]
/opt/databook/databook/physiology/ophys/visual-behavior/VBO-Tutorial-Compare_trial_types.md: WARNING: Notebook exception traceback saved in: /root/capsule/results/_build/html/reports/physiology/ophys/visual-behavior/VBO-Tutorial-Compare_trial_types.err.log [mystnb.exec]

/opt/databook/databook/physiology/stimuli/visual-behavior/VB-BehaviorSessionData.md: WARNING: Executing notebook failed: CellExecutionError [mystnb.exec]
/opt/databook/databook/physiology/stimuli/visual-behavior/VB-BehaviorSessionData.md: WARNING: Notebook exception traceback saved in: /root/capsule/results/_build/html/reports/physiology/stimuli/visual-behavior/VB-BehaviorSessionData.err.log [mystnb.exec]
```

### Stack trace
This one from `VBO-ExperimentData.md`:
```python
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In[42], line 10
      8 # add the colors for each image to the stimulus presentations table in the dataset
      9 stimulus_presentations = ophys_experiment.stimulus_presentations
---> 10 stimulus_presentations['color'] = ophys_experiment.stimulus_presentations['image_name'].map(lambda image_name: colormap[image_name])

File /opt/envs/allensdk/lib/python3.8/site-packages/pandas/core/series.py:4539, in Series.map(self, arg, na_action)
   4460 def map(
   4461     self,
   4462     arg: Callable | Mapping | Series,
   4463     na_action: Literal["ignore"] | None = None,
   4464 ) -> Series:
   4465     """
   4466     Map values of Series according to an input mapping or function.
   4467 
   (...)
   4537     dtype: object
   4538     """
-> 4539     new_values = self._map_values(arg, na_action=na_action)
   4540     return self._constructor(new_values, index=self.index).__finalize__(
   4541         self, method="map"
   4542     )

File /opt/envs/allensdk/lib/python3.8/site-packages/pandas/core/base.py:890, in IndexOpsMixin._map_values(self, mapper, na_action)
    887         raise ValueError(msg)
    889 # mapper is a function
--> 890 new_values = map_f(values, mapper)
    892 return new_values

File /opt/envs/allensdk/lib/python3.8/site-packages/pandas/_libs/lib.pyx:2924, in pandas._libs.lib.map_infer()

Cell In[42], line 10, in <lambda>(image_name)
      8 # add the colors for each image to the stimulus presentations table in the dataset
      9 stimulus_presentations = ophys_experiment.stimulus_presentations
---> 10 stimulus_presentations['color'] = ophys_experiment.stimulus_presentations['image_name'].map(lambda image_name: colormap[image_name])

KeyError: nan
```

### PR DESCRIPTION
**Pull request by JEElsner**
Fix #84 

When sorting an array of strings, NumPy turns NaN floats into the string 'nan'. This breaks the map from `stimulus_presentation['image_name']` since that column still has the NaN values. So we cast those values to a string.

### CODE REVIEW

**Comment by JEElsner** [comment]
Note that `databook/physiology/stimuli/visual-behavior/VB-BehaviorSessionData.md` was moved to `databook/physiology/ophys/visual-behavior/VB-BehaviorSessionData.md`, so that will need to be fixed in the merge.

**Comment by matchings** [comment]
So technically this should not be an issue if the fixes in #41  had been implemented already, as #41 modifies these notebooks to sub-select `stimulus_block_name` = `change_detection_behavior` prior to doing anything with the `image_name`, and there are no NaN values in the `image_name` after selecting the `change_detection_behavior` block. 

So i apologize for #41 not being implemented already. However the fix in this PR also works and is fine by me to merge.

**Review by matchings — APPROVED** [review]
As i noted in comments on the PR, the issue with `image_name` including NaNs should not occur if the `change_detection_behavior` stimulus block is first selected, which is addressed in a separate PR. However it doesnt hurt anything to cast to a string here, so this can be merged without any issue.

**Comment by JEElsner** [comment]
Okay let's hold off on merging this for now then until #41 gets merged. As long as that fixes everything, I'll delete this one. 👍

**Pull request closed without merging**
